.lgtm/cpp-queries/fgets.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Use of fgets()
+ * @description fgets() is dangerous to call. Use read_line() instead.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id cpp/fgets
+ * @tags reliability
+ *       security
+ */
+import cpp
+
+
+predicate dangerousFunction(Function function) {
+  exists (string name | name = function.getQualifiedName() |
+    name = "fgets")
```

Comment:
hmm, we might want to list more here actually, specifically, reading through CODING_STYLE:

* htons(), ntohs(), htonl(), ntohl()
* dup()
* atoi(), atol(), atoll()

Also, the "ENOTSUP" error should not be used (as the correct name is EOPNOTSUPP...)

Reply:
@poettering I extended the filter so that it would match the functions you mentioned. The results can be seen at https://lgtm.com/query/1506717996901/. If that's OK I can open a PR to roll it out.

It's probably worth mentioning that `htonl` is used in `IN6_IS_ADDR_LINKLOCAL` and we most likely can't do anything about it, so we might need to silence the alert with the special comments ([that aren't in line with our coding style](https://discuss.lgtm.com/t/rfe-support-for-supression-comments-in-comment-blocks/1364)) or simply ignore them :-)

Reply:
looks great. Though "deprecated" is not the best word, that's up for glibc to decide. Let's just say "please don't use it in our sources" or so...

I mean, there *are* ways to use fgets() properly, and htonl() vs htobe32() is just a matter of taste, there's no technical reason to prefer one over the other, they are exactly the same thing, the latter is just so much better named, since it doesn't pretend longs were 32bit...

Reply:
Naming things in English isn't my best skill :-) I'll try to come up with something clearer on Monday.

By the way, it'd be great if LGTM could post GIFs. It will make my life a bit easier and PRs a bit funnier :-)

Reply:
emojis should work, no? 🎈🎉🏆🏅 if something good happens, and maybe 👿💀💣💩 if not? ;-)